### ISSUE TITLE
MatSelect initial value

### ISSUE THREAD

**Issue by ctrl-alt-d** [SamProf/MatBlazor#165]
MatSelect doesn't set initial value.

Expected:

When I set an initial value, I expect select will be rendered with this value selected:

```
<MatSelect Label="Pick a Food Group" @bind-Value="@Val">
    <MatOption Value=""></MatOption>
    <MatOption Value="grains">Bread, Cereal, Rice, and Pasta</MatOption>
    <MatOption Value="vegetables">Vegetables</MatOption>
    <MatOption Value="fruit">Fruit</MatOption>
</MatSelect>

@functions
{
    public string Val {get; set; } = "vegetables";  //<----- initial value
}
```

But, Matblazor, Always has "first value" selected:

https://blazorfiddle.com/s/ns9i5c5s

**Comment by cs2090** [comment]
I identified the same issue.  As a work around in I replaced ```<MatOption>``` with ```<Option>``` and it appears to work ok.   
```
<MatSelect Label="Pick a Food Group" @bind-Value="@Val">
     <Option Value=""></Option>
     <Option Value="grains">Bread, Cereal, Rice, and Pasta</Option>
     <Option Value="vegetables">Vegetables</Option>
     <Option Value="fruit">Fruit</Option>
</MatSelect>
 
@code
{
    public string Val {get; set; } = "vegetables";
 }
```

A further curiosity is if I leave the first option value empty (```<Option Value=""></Option>```),  the ```<MatTextField>``` following the ```<MatSelect>``` seems to be misaligned.  

``` 
<MatSelect Label="Pick a Food Group" @bind-Value="@Val" Style="width:45%;">
     <Option Value=""></Option>
     <Option Value="grains">Bread, Cereal, Rice, and Pasta</Option>
     <Option Value="vegetables">Vegetables</Option>
     <Option Value="fruit">Fruit</Option>
</MatSelect>
<MatTextField @bind-Value="@AddVal" Style="width:45%;"/>         
@code
{
     public string Val {get; set; }
     public string AddVal {get; set; }
}
```
Changing ```<Option Value=""></Option>``` to  ```<Option Value=" ">Please select a Food Group</Option>``` eliminates the issue.  Note the value is a single space (" ") in the revised code.

Great work on MatBalzor to all involved.  I am in the process of attempting something beyond just playing around with Blazor and this framework is saving me a lot of effort on the design side.

**Comment by ctrl-alt-d** [comment]
Thanks @cs2090 I will try.

**Comment by ArryMatt** [comment]
Amazing @cs2090, that saved me a load of code.

**Comment by ctrl-alt-d** [comment]
Workaround from `@Khenzar` on gitter [https://blazorfiddle.com/s/grqktq64](https://blazorfiddle.com/s/grqktq64)  using `attributes`:

```
@page "/"
 
        <MatSelect Label="Pick a Food Group" @bind-Value="@Val">
            <MatOption Value=""></MatOption>
            <MatOption @attributes="@(GetSelected(Val=="grains"))" Value="grains">Bread, Cereal, Rice, and Pasta</MatOption>
            <MatOption @attributes="@(GetSelected(Val=="vegetables"))" Value="vegetables">Vegetables</MatOption>
            <MatOption @attributes="@(GetSelected(Val=="fruit"))" Value="fruit">Fruit</MatOption>
        </MatSelect>
 
        @functions
        {
            public string Val {get; set; } = "vegetables";
            Func<bool, Dictionary<string, object>> GetSelected = selected => selected ? new Dictionary<string, object>() { { "selected", true } } : null;
        }
```

**Comment by SamProf** [comment]
Thanks for your issue/idea. In the near future will be implemented.

**Comment by themiwi** [comment]
@ctrl-alt-d, thanks for the work-around. Have you managed to get it to work with `Enhanced="true"`?

**Comment by ctrl-alt-d** [comment]
hi @themiwi,

When  Enhanced is set to "true", matSelect doesn't use a `select` and the workaround is intended for `select` html tag. I guess it doesn't work.

**Comment by SamProf** [comment]
A already sad sometimes. Select enhanced realy not completed. I just start but complete it. Please, don't use it.
Right now I working on new date picker, then I will working on EditForm for all components maybe then on select enhanced. 
Please, SORRY. I have no time to make all as fast you want. 
Thank you for undastanding.
AlsoI have a lot of ideas about new components, but I can't, because I need to fix bugs.

**Comment by katiep23** [comment]
> A further curiosity is if I leave the first option value empty (`<Option Value=""></Option>`), the `<MatTextField>` following the `<MatSelect>` seems to be misaligned.

@cs2090, did you ever figure out why this is happening?  The workaround doesn't work correctly for me.  If I change it to <MatOption value="">Select</MatOption> instead of <MatOption value=""></MatOption>, it lines up properly though.  But it's not a route I really want to take.

**Comment by Paul-Schroeder** [comment]
Thanks for the clarifications @SamProf and workaround @ctrl-alt-d.  I am curious if this is still on the radar / any change in status on this one.  In the meantime, I was able to use the workaround so long as I set Enhanced="false".  

I took a quick look at the <div> based output from using Enhanced="true" with the idea of trying to develop a workaround for a pull request.  However, I came up empty...perhaps if one of you post thoughts/suggestions on how you might tackle this one someone from the community might try to pick it up?
